### DISCUSSION TITLE
Can't get Watchtower Notifications to work. Help? Synology/Docker/Notifiarr

### DISCUSSION THREAD

**Discussion by SawkeeReemo** [containrrr/watchtower#1804] [Q&A]
_**GOAL: For now I just want Watchtower to notify me when a container has an update, I don't want to automate the actual updates themselves. I'm using Notifiarr to link this to Discord.**_

_`trying to not sound as frustrated as I am haha`_ 
I've spent about 3 days trying to figure out how to set up Watchtower on my Synology NAS in a Docker container, and I can't find any guides out there (_yet_) that give me results that work. My latest attempt follows this guide:
[https://drfrankenstein.co.uk/2021/12/31/watchtower-automated-docker-updates-on-a-synology-nas/](https://drfrankenstein.co.uk/2021/12/31/watchtower-automated-docker-updates-on-a-synology-nas/)

It didn't work, so I just started guessing at what I'm supposed to put in the docker-compose file, and this is where I ended up before deciding to give up and just post in here:
![Screen Shot 2023-10-18 at 7 53 01 PM](https://github.com/containrrr/watchtower/assets/60272670/fb406004-35ab-49d7-b9d9-03e60d0e2341)
_(I commented out the command items just as a placeholder. Once I get this to work, I may decide to use some of those.)_

Even though you can see my settings above, when it runs, the logs just show:
![Screen Shot 2023-10-18 at 8 00 00 PM](https://github.com/containrrr/watchtower/assets/60272670/ea67078e-054f-49c2-933a-850f84d384e6)

The last line reads: `level=info msg="Session done" Failed=0 Scanned=24 Updated=18 notify=no`
Which is not what I'd expect. Yes, there are updates available, but no it did not update 18 items, and yes I have my `WATCHTOWER_NOTIFICATION_REPORT=true` set. So why did it ignore all that and then not actually do anything anyways, but report that it did?? I'm next-level confused.

Can anyone help please? I'd really appreciate it.

_**UPDATE 1:**_ I removed the PGID and PUID environmental variables from that compose file and rebuilt the container thinking that might be the problem. Nope, same non-results. I don't know what else to do at this point. Do I need to run the container with "high privileges" or something?

_**UPDATE 2:**_ Out of frustration, I decided to run the container with high privileges. Same thing happened, no notifications, false job report, me wanting to become Amish. 😜

**Comment by piksel** [comment]
Hello! Yeah, the updated in the log is a bit misleading and it counts "stale" (containers that *could* be updated) as well due to it being tied to metrics that does this for legacy reasons [^1]. It's not the case for the reports (notifications) though.

I think your problem is related to the `generic` configuration or payload. You should receive a notification on startup, so that is the easiest way to tell if it works. 

The `generic` notification handler doesn't know what to expect from the target service, so it doesn't report errors very well. 

[^1]: Typing it out, it's pretty obvious that this behaviour is really strange and that the log should use the report values instead of the metric ones.

**Reply by piksel** [reply]
There is an issue for notifiarr support:
https://github.com/containrrr/watchtower/issues/1523

I think what you need is to add:
```yaml
- WATCHTOWER_NOTIFICATION_TEMPLATE=json.v1
```
To the env variables.

Or maybe that isn't supported by notifiarr yet? In that case it should just be a matter of adding `?template=json` to the end of your notification URL.

**Reply by SawkeeReemo** [reply]
Hey! Thanks, and yes, I do receive the initial notification. But then nothing ever happens after that.
![IMG_5852](https://github.com/containrrr/watchtower/assets/60272670/0e6b8eed-8a7e-4663-bb8e-9d5f1dddf0ef)

**Reply by SawkeeReemo** [reply]
I tried adding that env variable to my compose, recreated the container, but now it’s trying to connect to shoutrrr instead, even though nothing exists at all for that. And now I don’t get the initial launch notification at all. So, one step backwards, unfortunately. 

I’ll try adding that bit to the end of the webhook URL to see what happens next. 

But also, why does it say zero notifications? Wouldn’t it throw an error or something? Seems to me like it’s completely ignoring the notification env all together.

**Reply by SawkeeReemo** [reply]
Actually it’s already on there. ¯\_(ツ)_/¯ 

`- WATCHTOWER_NOTIFICATION_URL=generic://notifiarr.com/api/v1/notification/watchtower/[REDACTED]?template=json`

**Reply by piksel** [reply]
Yeah, okay. It's because the default notification template avoids sending notifications when there are no updated or failed containers. You can use the template example here to always send notifications:
https://containrrr.dev/watchtower/notifications/#:~:text=Example%20using%20a%20custom%20report%20template%20that%20always%20sends%20a%20session%20report%20after%20each%20run%3A

**Reply by SawkeeReemo** [reply]
I will give that a try. 

Quick question: Since I’m using the MONITOR_ONLY env, do you think that might somehow be affecting it? 

I’ve also spoken with the Notifiarr dev about this issue, and so far I think that’s the only difference I’ve seen in my testing. Curious to know what the trigger for this could be.

**Reply by SawkeeReemo** [reply]
Having trouble figuring out how to get this into a docker compose properly, their example doesn't work with my formatting. Trying to see if I can rewrite mine to match theirs... new territory for me though. I'll respond once/if I figure this out.

**Reply by SawkeeReemo** [reply]
I figured out how to rewrite my compose to include their example. 

Well, now I get a job end report notification as well, but it's still not telling me which containers need to be updated. It just tells me that it ran and that nothing was updated... but then no list of containers that need to be updated. It's also not running `WATCHTOWER_CLEANUP`.

**Reply by SawkeeReemo** [reply]
Another update, I activated the automatic updates on my test system, and those work properly as well as sending me a notification on Discord. But what doesn't work is receiving a notification when a container has an update, but does not get updated. Seems like a pretty big bug to me, or maybe there's some setting that we're just missing?

**Reply by piksel** [reply]
Yeah, the template I linked actually doesn't print `.Stale` containers. As you might have noticed, the report templates were not created with the `monitor-only` use case in mind. We should update the docs and default template to include them. 
If you only need to monitor, you can just replace `.Updated` with `.Stale` in the template and it should do what you want.

I have added a [template preview](https://containrrr.dev/watchtower/template-preview/) to the docs (but it's still in alpha quality, hence why it's not linked directly from the docs) where you can preview template changes.

**Reply by piksel** [reply]
I created two templates that should work well with monitor-only and put them here:
https://github.com/containrrr/watchtower/discussions/1806

To better help with this scenario for other users, I also added an example configuration in https://github.com/containrrr/watchtower/discussions/1807 based on your config (except for the `notifiarr` notifications which warrants it's own topic).

**Reply by SawkeeReemo** [reply]
Thank you. I’m still a bit of a newb in this area, like I think I understand some of the mechanics of what that big format array looking thing is, but I’m basically copying and pasting it, then crossing my fingers. Haha know what I mean? 

Thanks for sending it! As soon as I get some time to play around, I will drop this in to test.

**Reply by SawkeeReemo** [reply]
Am I able to use this in combination with `WATCHTOWER_LABEL_ENABLE=true` or a command list of containers I actually want to update? It'd be great to be able to keep updating containers that aren't critical, while also being notified about ones that have updates available. 

UPDATE: Gave it a shot, the answer is no for me so far. This was my compose: https://hastebin.com/share/vahavegevo.yaml

**Reply by piksel** [reply]
This should actually be possible now (landed in v1.6.0) by using [label-take-precedence](https://containrrr.dev/watchtower/arguments/#with_label_taking_precedence_over_arguments) and putting the 
`com.centurylinklabs.watchtower.monitor-only="false"` label on those containers ([docs](https://containrrr.dev/watchtower/container-selection/#monitor_only)).

Or if you want to do it the other way around, you can put the label `com.centurylinklabs.watchtower.monitor-only="true"` on the critical ones and remove `WATCHTOWER_MONITOR_ONLY` (or set it to false).

**Comment by austinwbest** [comment]
So i have worked with him some and can confirm he is getting the startup notification but absolutely nothing else when using the monitor only variable. He does get them when containers update as well.

Is it a bug that monitor only does not send anything even with that template being used or is it by design that when only monitoring it does not send notifications?

**Reply by SawkeeReemo** [reply]
And here is the docker-compose used to create the container for testing:
https://hastebin.com/share/uvokawuqiy.ruby

**Comment by Nectulus** [comment]
Bump.

I realize it's been a year since last post however I am having an exact same issue with generic notifications to use with Notifiarr. Watchtower is configured in in Monitor Only mode. Just like the original poster I am receiving the initial message when container is starting but nothing after this.

My compose file if this helps.
[](https://hastebin.com/share/opumufibam.ruby)

I found a few other posts here and on Reddit describing this issue. It appears that this only happens in Monitor Only mode which is exactly what I want.

Has this ever been resolved?

Thank you.

**Reply by austinwbest** [reply]
Sorry bud, can't help. I stopped using it shortly after this and moved to something else. I needed dependable notifications & the ability to set them on a container by container basis as well as check for updates and auto update container by container.

Even though I don't use it anymore if there is something needed notifiarr side I can still make the changes.

Best of luck!

**Reply by SawkeeReemo** [reply]
Same. I switched to another app by the dev of notifiarr called Dockwatch. Easier to configure and built to work with notifiarr.  You should grab that.

**Reply by Nectulus** [reply]
Just found it and set it up, thank you.
